cmd/cmd.go

```diff
@@ -0,0 +1,127 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+
+	"github.com/axiomhq/axiom-go/axiom"
+	"go.uber.org/zap"
+
+	"github.com/axiomhq/pkg/version"
+)
+
+// MainFunc is implemented by the main packages and passed to the `Run` function
+// which takes care of signal handling, loading the runtime configuration and
+// setting up logging, the Axiom client, etc. It must block until the context
+// is marked done. Errors returned from the `MainFunc` should be created using
+// the `Error()` function.
+type MainFunc func(context.Context, *Factory) error
+
+// Run the named app with the given `MainFunc`. Additionally, options can be
+// passed to configure the behaviour of the bootstrapping process.
+func Run(appName string, fn MainFunc, options ...Option) {
+	if code := run(appName, fn, options...); code != exitOK {
+		code.exit()
+	}
+}
+
+func run(appName string, fn MainFunc, options ...Option) exitCode {
+	// Setup the default config and apply the supplied options.
+	cfg := &config{
+		loggerOptions: DefaultLoggerOptions(),
+		signals:       DefaultSignals(),
+	}
+	for _, option := range options {
+		if err := option(cfg); err != nil {
+			return exitConfig
+		}
+	}
+
+	// Set up logger.
+	var (
+		logger *zap.Logger
+		err    error
+	)
+	if v := os.Getenv("DEBUG"); v != "0" {
```

Comment:
I'd rather we default to production logger and check for `"1"` (or event `strconv.ParseBool`)

Reply:
Agreed.